### PR TITLE
Convert empty string to null string when loading

### PR DESCRIPTION
**Pull request by LokQiJun**
Add checks to string obtained from xml file to determine if it is an empty string.
Convert these empty string found to a null string
Added assertions to ensure compulsory values are not null

### CODE REVIEW

File: src/main/java/seedu/moneygowhere/storage/LocalStorage.java [hunk 1]
```diff
@@ -197,18 +198,35 @@ private Expense createExpense(Node node) throws LocalStorageLoadDataInputError {
...
         String remark = element.getElementsByTagName(XML_EXPENSE_REMARKS_ELEMENT)
                 .item(0).getTextContent();
+        if (remark.isEmpty() || remark.trim().isEmpty()) {
+            remark = null;
+        }
+
+        NodeList amountNodeList = element.getElementsByTagName(XML_EXPENSE_AMOUNT_ELEMENT);
+        String currency = amountNodeList.item(0).getAttributes()
+                .getNamedItem(XML_EXPENSE_AMOUNT_CURRENCY_ATTRIBUTE).getTextContent();
+        assert !(currency.isEmpty() || currency.trim().isEmpty()) : "There must be a currency";
```
Comment (xzynos) [inline src/main/java/seedu/moneygowhere/storage/LocalStorage.java:224]: May I ask is there a reason why assert is used to perform input validation? I don't believe assert is enabled by default in most cases.

**Review by xzynos — COMMENTED** [review]
May I ask is there a reason why assert is used instead of exceptions to validate the data files?

File: src/main/java/seedu/moneygowhere/storage/LocalStorage.java (hunk 1 above)
Comment (LokQiJun) [reply src/main/java/seedu/moneygowhere/storage/LocalStorage.java]: Sorry this was a mistake on my part. Will move assertions into constructor of the objects and use exception in the loading of data.

**Comment by jeyvia** [comment]
Looks good, will merge.

**Pull request merged**